src/components/numeral-date/numeral-date.test.tsx

```diff
@@ -0,0 +1,1995 @@
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+    render(
+      <CarbonProvider validationRedesignOptIn>
+        <NumeralDate
+          value={{ dd: "31", mm: "", yyyy: "2020" }}
+          onChange={() => {}}
+          enableInternalError
+        />
+      </CarbonProvider>
+    );
+
+    const monthInput = screen.getByRole("textbox", { name: "Month" });
+    const dayInput = screen.getByRole("textbox", { name: "Day" });
+    const yearInput = screen.getByRole("textbox", { name: "Year" });
+    await user.type(monthInput, "0");
+    await user.tab();
+
+    expect(dayInput).toBeInvalid();
+    expect(monthInput).toBeInvalid();
+    expect(yearInput).toBeInvalid();
+    expect(
+      screen.getByText("Month should be a number within a 1-12 range.")
+    ).toBeVisible();
+  });
+
+  it("should render the validation message when the `Year` is blurred and has a value less than 1800", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+    render(
+      <CarbonProvider validationRedesignOptIn>
+        <NumeralDate
+          value={{ dd: "31", mm: "", yyyy: "" }}
+          onChange={() => {}}
+          enableInternalError
+        />
+      </CarbonProvider>
+    );
+
+    const yearInput = screen.getByRole("textbox", { name: "Year" });
+    const dayInput = screen.getByRole("textbox", { name: "Day" });
+    const monthInput = screen.getByRole("textbox", { name: "Month" });
+    await user.type(yearInput, "1799");
+    await user.tab();
+
+    expect(dayInput).toBeInvalid();
+    expect(monthInput).toBeInvalid();
+    expect(yearInput).toBeInvalid();
+    expect(
+      screen.getByText("Year should be a number within a 1800-2200 range.")
+    ).toBeVisible();
+  });
+
+  it("should render the validation message when the `Year` is blurred and has a value greater than 2200", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+    render(
+      <CarbonProvider validationRedesignOptIn>
+        <NumeralDate
+          value={{ dd: "31", mm: "", yyyy: "" }}
+          onChange={() => {}}
+          enableInternalError
+        />
+      </CarbonProvider>
+    );
+
+    const yearInput = screen.getByRole("textbox", { name: "Year" });
+    const dayInput = screen.getByRole("textbox", { name: "Day" });
+    const monthInput = screen.getByRole("textbox", { name: "Month" });
+    await user.type(yearInput, "2201");
+    await user.tab();
+
+    expect(dayInput).toBeInvalid();
+    expect(monthInput).toBeInvalid();
+    expect(yearInput).toBeInvalid();
+    expect(
+      screen.getByText("Year should be a number within a 1800-2200 range.")
+    ).toBeVisible();
+  });
+
+  it("should render the appropriate validation message when the `Day` and `Month` fields are blurred and have values greater than the valid limits", async () => {
```

Comment:
**comment** (non-blocking): I don't personally see the need for checking every combination of 2 of 3 fields with errors for the specific combined error message. I think just one test, ideally with errors in all 3, should give us enough confidence that all the individual errors are rendered. Maybe you need to do it this way for coverage though (I haven't checked)?

Reply:
There is some coverage concerns here but I could definitely reduce the set I have here. In the long run I'd prefer to move these to chromatic and use play functions but in the interim I'd prefer to be a little overly cautious in testing them as each input works independently and we want to guarantee the order of the messages when in the old validation pattern etc